catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/RemoteDeleteOperations.java

```diff
@@ -0,0 +1,127 @@
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.Constants;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.impl.FrameworkProperties;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.DeleteResponse;
+import ddf.catalog.operation.ProcessingDetails;
+import ddf.catalog.operation.impl.DeleteResponseImpl;
+import ddf.catalog.operation.impl.ProcessingDetailsImpl;
+import ddf.catalog.source.CatalogStore;
+import ddf.catalog.source.IngestException;
+
+public class RemoteDeleteOperations {
+
+    static final Logger INGEST_LOGGER = LoggerFactory.getLogger(Constants.INGEST_LOGGER_NAME);
+
+    private OperationsCatalogStoreSupport opsCatStoreSupport;
+
+    private FrameworkProperties frameworkProperties;
+
+    private OperationsMetacardSupport opsMetacardSupport;
+
+    public RemoteDeleteOperations(FrameworkProperties frameworkProperties,
+            OperationsMetacardSupport opsMetacardSupport,
+            OperationsCatalogStoreSupport opsCatStoreSupport) {
+
+        this.opsCatStoreSupport = opsCatStoreSupport;
+        this.frameworkProperties = frameworkProperties;
+        this.opsMetacardSupport = opsMetacardSupport;
+
+    }
+
+    DeleteResponse performRemoteDelete(DeleteRequest deleteRequest, DeleteResponse deleteResponse) {
+
+        if (!opsCatStoreSupport.isCatalogStoreRequest(deleteRequest)) {
+            return deleteResponse;
+        }
+
+        DeleteResponse remoteDeleteResponse = doRemoteDelete(deleteRequest);
+        if (deleteResponse == null) {
+            deleteResponse = remoteDeleteResponse;
+            deleteResponse = injectAttributes(deleteResponse);
+        } else {
+            deleteResponse.getProperties()
+                    .putAll(remoteDeleteResponse.getProperties());
+            deleteResponse.getProcessingErrors()
+                    .addAll(remoteDeleteResponse.getProcessingErrors());
+        }
+        return deleteResponse;
+    }
+
+    private DeleteResponse doRemoteDelete(DeleteRequest deleteRequest) {
+        HashSet<ProcessingDetails> exceptions = new HashSet<>();
+        Map<String, Serializable> properties = new HashMap<>();
+
+        List<CatalogStore> stores = opsCatStoreSupport.getCatalogStoresForRequest(deleteRequest,
+                exceptions);
+
+        List<Metacard> metacards = new ArrayList<>();
+        for (CatalogStore store : stores) {
+            try {
+                if (!store.isAvailable()) {
+                    exceptions.add(new ProcessingDetailsImpl(store.getId(),
+                            null,
+                            "CatalogStore is not available"));
+                } else {
+                    DeleteResponse response = store.delete(deleteRequest);
+                    properties.put(store.getId(), new ArrayList<>(response.getDeletedMetacards()));
+                    metacards = response.getDeletedMetacards();
```

Comment:
I agree with Joe and Ryan. Should be the full set of deleted metacards, not this last-in-wins error.